src/taker/api.rs

```diff
@@ -241,6 +246,52 @@ impl Taker {
     /// respond back, the swap round will fail.
     #[tokio::main]
     pub async fn send_coinswap(&mut self, swap_params: SwapParams) -> Result<(), TakerError> {
+        let tor_log_dir = "/tmp/tor-rust-taker/log".to_string();
+
+        let taker_port = self.config.port;
+        let taker_socks_port = self.config.socks_port;
+
+        if Path::new(tor_log_dir.as_str()).exists() {
+            match fs::remove_file(Path::new(tor_log_dir.clone().as_str())) {
+                Ok(_) => log::info!("Previous taker log file deleted successfully"),
+                Err(_) => log::error!("Error deleting taker log file "),
+            }
+        }
+
+        thread::sleep(Duration::from_secs(10));
+
+        let handle = mitosis::spawn(
+            (taker_socks_port, taker_port),
+            |(taker_socks_port, taker_port)| {
+                let _handler = Tor::new()
+                    .flag(TorFlag::DataDirectory("/tmp/tor-rust-taker/".into()))
+                    .flag(TorFlag::LogTo(
+                        LogLevel::Notice,
+                        LogDestination::File("/tmp/tor-rust-taker/log".to_string()),
+                    ))
+                    .flag(TorFlag::SocksPort(taker_socks_port))
+                    .flag(TorFlag::HiddenServiceDir(
+                        "/tmp/tor-rust-taker/hs-dir/".into(),
+                    ))
```

Comment:
I feel like tor spawning should be an independent taker API. Should be the first thing called at `send-coinswap`, and hard error if tor isn't fully up till the `monitor_log_for_completion`.